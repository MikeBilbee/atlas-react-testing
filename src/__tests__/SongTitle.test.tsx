//SongTitle.test.tsx
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import SongTitle from '../components/SongTitle';

describe('SongTitle component', () => {
  it('renders with title and artist', () => {
    const { container } = render(<SongTitle title="Test Title" artist="Sample Artist" />);
    expect(container).toMatchSnapshot();
  });

  it('renders with long title and artist', () => {
    const longTitle = 'This is a very long song title that might wrap to multiple lines';
    const longArtist = 'This is also a very long artist name that might wrap';
    const { container } = render(<SongTitle title={longTitle} artist={longArtist} />);
    expect(container).toMatchSnapshot();
  });
});