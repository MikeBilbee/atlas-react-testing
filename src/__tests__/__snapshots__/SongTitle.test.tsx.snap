// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle component > renders with long title and artist 1`] = `
<div>
  <div>
    <h1
      class="text-3xl font-bold dark:text-gray-400 mb-2"
    >
      This is a very long song title that might wrap to multiple lines
    </h1>
    <h2
      class="text-gray-600 dark:text-gray-500"
    >
      This is also a very long artist name that might wrap
    </h2>
  </div>
</div>
`;

exports[`SongTitle component > renders with title and artist 1`] = `
<div>
  <div>
    <h1
      class="text-3xl font-bold dark:text-gray-400 mb-2"
    >
      Test Title
    </h1>
    <h2
      class="text-gray-600 dark:text-gray-500"
    >
      Sample Artist
    </h2>
  </div>
</div>
`;
